woocommerce/woocommerce-gateway-stripe
# Remove handling of the removed statement_descriptor and short_statement_descriptor settings

## Changes proposed in this Pull Request:

- Remove the saving of the `statement_descriptor` and `short_statement_descriptor` settings from the admin side.
- Remove the displaying of error messages for these fields on the admin side.
- Remove the instructions and string-length counter from their input field.

These fields have been read-only since 7.9.2, introduced in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2833. Since then. Since then, we've retrieved the values from these fields from the Stripe account data.

What still needs to be done but isn't handled in this PR:
- These settings are still saved in the database when coming from all installs. This PR doesn't intend to remove those settings from the database.
- We're still using these settings for sources when processing payments. To keep the PRs and testing small, I'm planning to handle that in a separate PR.

## Testing instructions

Check for regressions when paying with cards + Full bank statement descriptor

1. Go to the Stripe settings page > Settings tab at `siteurl/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=settings`
2. Disable `Add customer order number to the bank statement` and save
3. Confirm there are no errors in the browser console or PHP logs
4. Check out with a card using the Legacy experience
5. Check out with a card using the Updated experience
6. Go to the Stripe dashboard https://dashboard.stripe.com/test/payments
7. Open the payment details for the last two purchases
8. Scroll down to "Payment details"
9. Confirm that the value for "Statement descriptor" matches the preview Stripe settings page > Settings tab > Customer bank statement > Preview

Check for regressions when paying with cards + Short bank statement descriptor
1. Go to the Stripe settings page > Settings tab at `siteurl/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=settings`
2. Enable `Add customer order number to the bank statement` and save
3. Confirm there are no errors in the browser console or PHP logs
4. Check out with a card using the Legacy experience
5. Check out with a card using the Updated experience
6. Go to the Stripe dashboard https://dashboard.stripe.com/test/payments
7. Open the payment details for the last two purchases
8. Scroll down to "Payment details"
9. Confirm that the value for "Statement descriptor" matches the preview Stripe settings page > Settings tab > Customer bank statement > Preview

---


**Post merge**
